### DISCUSSION TITLE
Can I include unreleased content at the top as "pre-release" notes without discarding them?

### DISCUSSION THREAD

**Discussion by sheldonhull** [miniscruff/changie#488] [Q&A] [answered]
If I'm not yet releasing, and have pending changes in my trunk, can I include those in the changelog without getting rid of the fragments?

**Comment by miniscruff** [comment]
`changie batch version|major|minor|patch|auto --keep`, this one?

**Reply by sheldonhull** [reply]
I tried, but it's not quite the same.
It increments and adds as a new version to the changelog. I don't want to increment the version yet, because I actually driven releases from this tool. 

Instead I want the equivalent of:


```shell

changie merge --include-unreleased
```

that would result in something like the follow fragment for new product feature, but include a collection of all the pending changes.

Could use a precommit hook to make sure I keep the changelog updated with all the changes coming up for folks.


```markdown

# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html),
and is generated by [Changie](https://github.com/miniscruff/changie).

## unreleased

### 🎉 New Product Feature

- Add this foo...
  Not tagging yet, as this is untested, so keeping as prerelease notes.

## v0.1.0 - 2023-05-18

### 🤖 CI & Build

- Trigger this job anytime I push to main, so I can verify behavior with less pain 😀.

```

**Reply by miniscruff** [reply]
Oh, I see what you are saying, currently no that is not one of the options. For me I only ever change the changelog on release, as it is my release trigger, so this would not have come up for me.

It is also possible to use `--keep` and/or `--move-dir` and create a "theoretical" version you mark as unreleased. Something like...

```sh
# on pr merge when not our gh-bot account
changie batch patch --meta unreleased --keep
changie merge
```

But it shouldn't be too hard to add the `--include-unreleased`, can create an issue and see if I get time for it. Or someone else can give it a try. Just need to make an "unreleased header" config option that can be defaulted to `""`.

**Reply by miniscruff** [reply, marked as answer]
Closing in favor of this issue: https://github.com/miniscruff/changie/issues/503
